flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1379,15 +1399,17 @@ struct XEmboxOpConversion : public EmboxCommonConversion<fir::cg::XEmboxOp> {
       if (hasSlice)
         sliceOff += 3;
     }
-    auto hasSubcomp = !xbox.subcomponent().empty();
     if (hasSlice || hasSubcomp) {
       llvm::SmallVector<mlir::Value> args = {base, ptrOffset};
       args.append(gepArgs.rbegin(), gepArgs.rend());
       if (hasSubcomp) {
         // For each field in the path add the offset to base via the args list.
-        // In the most general case, some offsets must be computed since they
-        // are not be known until runtime.
-        TODO(loc, "intra-entity slice in fir.embox codegen");
+        // TBD: In the most general case, some offsets must be computed since
```

Comment:
I wasn't sure what this comment meant, so I added aTBD. When are offsets not known until runtime?